perun-base/src/main/java/cz/metacentrum/perun/core/api/BeansUtils.java

```diff
@@ -617,6 +617,37 @@ public static String getPropertyFromCustomConfiguration(String propertyFile, Str
 		}
 
 	}
+
+	/**
+	 * True if this instance of perun is read only.
+	 * False if not.
+	 *
+	 * @return true or false (readOnly or not)
+	 */
+	public static boolean isThisPerunReadOnly() {
+		String readOnly;
+		try {
+			readOnly = BeansUtils.getPropertyFromConfiguration("perun.readOnlyPerun");
+		} catch (Exception ex) {
+			//If something wierd happens, set this to normal configuration (not readOnly) and log this exception
+			log.error("Problem occures when trying to get readOnly configuration from perun properties file.", ex);
+			log.debug("Read only configuration not found, set to false.");
+			return false;
+		}
+
+		if(readOnly == null) {
+			log.debug("Read only configuration is null, set to false.");
+			return false;
+		}
+
+		if(readOnly.contains("true")) {
+			log.debug("Read only configuration found='{}', set to true.", readOnly);
+			return true;
+		}
+
+		log.debug("Read only configuration found='{}', set to false.", readOnly);
```

Comment:
I'm not sure, how many times is ``PerunBasicDataSource.getConnection()`` called, but it will probably spawn log with debug message since it check if perun is read only everytime.